src/medicine/medicine.py

```diff
@@ -0,0 +1,70 @@
+        :param int|float dose: The dose (in mg) in a single pill
+        :param int|float daily_dose: The dose (in mg) to be taken daily
+        :param int current_amount: the current amount of pills available
+        """
+        if not isinstance(name, str):
+            raise ValueError('name is supposed to be a string')
+        self.name = name
+
+        if not isinstance(dose, (int, float)):
+            raise ValueError('dose is supposed to be a number')
+        if dose < 0:
+            raise ValueError('Negative doses do not exist')
+        self.dose = dose
+
+        if not isinstance(daily_dose, (int, float)):
+            raise ValueError('daily_dose is supposed to be a number')
+        if dose < 0:
+            raise ValueError('Negative doses do not exist')
+        self.daily_dose = daily_dose
+
+        if not isinstance(current_amount, int):
+            raise ValueError('current_amount is supposed to be an integer')
+        if current_amount < 0:
+            raise ValueError('It is not possible to have negative amounts of pills')
+        self.current_amount = current_amount
+
+    def serialize(self):
+        return {'name': self.name, 'dose': self.dose, 'daily_dose': self.daily_dose, 'current_amount': self.current_amount}
+
+    @property
+    def current_mg(self):
+        """Get the total amount of remaining medicine, in mg."""
+        return self.current_amount * self.dose
+
+    @property
+    def daily_pills(self):
+        # TODO: To nie koniecznie dobrze zadziała. Trzeba rozważyć co się stanie jeżeli:
+        # dawka dzienna nie dzieli się bez reszty (np. codziennie ma być brane 12mg, a pigułka ma 5mg)
+        return self.daily_dose / self.dose
+
+    @property
+    def days_left(self):
+        """Get the remaining number of days before the medicine runs out."""
+        # TODO: to będzie zwrócone jako ułamkowa liczba, co niekoniecznie jest to co byś chciał,
+        # return round(self.current_mg / self.daily_dose)  # Jeżeli ma być zaokrąglane
+        # return self.current_mg // self.daily_dose  # Jeżeli ma być zaokrąglane w dół
+        return self.current_mg / self.daily_dose
+
+    def add_pills(self, amount: int):
+        self.current_amount += amount
+
+    def take_dose(self):
+        # TODO: Jak obsłuzyć braki?
+        # jak jest mniej pigułek niż trzeba, to wezmniejsz tylko tyle ile jest. Inna opcja,
+        # to jeżeli nie starczy ci, to poprostu tego dnia nie bierzesz
+        pills = min(self.current_amount, self.daily_pills)
+        self.current_amount -= pills
+
+    def __str__(self):  # __str__() i __repr__() używam z przyzwyczajenia, tak mi wpojono. W docstringu masz co robią.
+        return f'{self.name.title()} with dose {self.dose} mg per pill (daily dose {self.daily_dose} mg) has {self.current_amount} pills left.'
+
+    def __repr__(self):
+        return f'Medicine({self.name}, {self.dose}, {self.daily_dose}, {self.current_amount})'
```

Comment:
wywaliłem te metody które zapisują do/z pliku, i są przeniesione do głównego pliku. 
Promuje się podejście do obiektówki gdzie wszystko się wali do klas. To czasem jest spoko, kiedy możesz rzeczywiście całość interakcji zamknąć w ramach tej jednej klasy, np. interakcje z bazą danych, ale to nie jest normalny stan rzeczy. Większość klas nie da się tak łatwo wyabstarchować, i potem te abstrakcje wychodzą w różnych dziwnych miejscach. Także jak masz klasy jak te dla twoich leków, to lepiej jest żeby były jak najmniejsze - tylko ich stan (tj. zmienne) i funckje które działają tylko i wyłącznie na tym stanie (czyli zmieniają go albo zwracają coś wyliczany na ich podstawie). Rzeczy typu zapis do pliku itp. nie zajmują się tylko tą klasą, bo ich główny adresat to plik. 
To nie jest twarda zasada. Rób żeby ci było wygodniej. Jest to poprostu dobry nawyk